app/lib/actions/candlepin/owner/import.rb

```diff
@@ -11,7 +11,11 @@ class Import < Candlepin::Abstract
 
         def run
           options = input.slice(:force, :upstream)
-          output[:response] = ::Katello::Resources::Candlepin::Owner.import(input[:label], input[:path], options)
+          cp_response = JSON.parse(
+            ::Katello::Resources::Candlepin::Owner.import(input[:label], input[:path], options)
+          )
+          output[:task] = cp_response # this also sets external_task
+          output[:task_id] = cp_response['id']
```

Comment:
It looks like you're taking the cadlepin task  data and passing it to AsyncImport, but all this can be done within this one action.  If you just make Import inherit from Candlepin::AbstractAsyncTask,  rename the run method to 'invoke_external_task', and then have it return the task hash, it should all be done in one action with no need to pass it around

Reply:
ahh i see you modeled this after the hypervisors task.  Its kind of a special case, i would probably more model this after all of the pulp actions, which just initiate and poll in the same action

Reply:
@jlsherrill I originally had it like that but it wasn't actually initiating polling.  I figured out that it's because `Actions::Base::Polling` (aka Dynflow::Polling) defines the `run` method which is where the [first call of `invoke_external_task`](https://github.com/Dynflow/dynflow/blob/e47118cd6bc6f972bec38844a6b06dd8b0739cca/lib/dynflow/action/polling.rb#L19) is, so overriding the `run` method would prevent polling from ever starting.  That's why I had separated it into the two actions.

Reply:
right, but if you change it to invoke_external_task, it should work and you can do it within a single action, similar to: https://github.com/Katello/katello/blob/master/app/lib/actions/pulp3/repository/sync.rb#L11-L14

Reply:
> rename the run method to 'invoke_external_task'

Whoops, glossed right over that part. :see_no_evil: I'll give it a try